packages/multichain/src/scope/types.ts

```diff
@@ -0,0 +1,103 @@
+import MetaMaskOpenRPCDocument from '@metamask/api-specs';
+import {
+  isCaipNamespace,
+  isCaipChainId,
+  parseCaipChainId,
+} from '@metamask/utils';
+import type {
+  CaipChainId,
+  CaipReference,
+  CaipAccountId,
+  KnownCaipNamespace,
+  CaipNamespace,
+  Json,
+} from '@metamask/utils';
+
+export enum KnownWalletScopeString {
+  Eip155 = 'wallet:eip155',
+}
+
+export type NonWalletKnownCaipNamespace = Extract<
+  KnownCaipNamespace,
+  KnownCaipNamespace.Eip155
+>;
+
+export const KnownWalletRpcMethods: string[] = [
+  'wallet_registerOnboarding',
+  'wallet_scanQRCode',
+];
+const WalletEip155Methods = ['wallet_addEthereumChain'];
```

Comment:
It seems that by using `string[]` we are discarding type information that could be valuable later. For instance, if we wanted a type from this array we couldn't do that easily. Thoughts on removing and using `as const`?

``` suggestion
export const KnownWalletRpcMethods = [
  'wallet_registerOnboarding',
  'wallet_scanQRCode',
] as const;
const WalletEip155Methods = ['wallet_addEthereumChain'] as const;

Reply:
makes the types too narrow and difficult to consume downstream. Same problem as https://github.com/MetaMask/core/pull/4784#discussion_r1813116599